src/NumericInput/index.js

```diff
@@ -49,11 +50,17 @@ class NumericInput extends TextInput {
         if (Number.isFinite(args.step)) {
             this._step = args.step;
         } else if (Number.isFinite(args.precision)) {
-            this._step = 1 / Math.pow(10, args.precision);
+            this._step = 10 / Math.pow(10, args.precision);
         } else {
             this._step = 1;
         }
```

Comment:
I'm a bit iffy about making this change as it's set to be 10x to what it was before (as the old behaviour was to hold shift to scale the step by 10 and now we are doing the opposite).

At the same time, I'm tempted to have this set to 1 or 0.1 as a default rather than using the precision just to have a know value over a relative one.